storm-client/src/jvm/org/apache/storm/daemon/supervisor/ClientSupervisorUtils.java

```diff
@@ -71,7 +71,9 @@ public static int processLauncherAndWait(Map<String, Object> conf, String user,
         try {
             process.waitFor();
         } catch (InterruptedException e) {
-            LOG.info("{} interrupted.", logPreFix);
+            LOG.warn("{} interrupted.", logPreFix);
+            Thread.currentThread().interrupt();
+            throw new IOException(logPreFix + " interrupted", e);
```

Comment:
Need a process.destroy() here before throwing the IOException, so the cleanup can happen.